### PR TITLE
Send mail to applicant

### PR DESCRIPTION
**Pull request by Akshiteshnaithani**
Targets #3083

### Description
New applicant is added to the Code Trek an email triggered to notify the applicant about their successful register.

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/CodeTrek/Services/CodeTrekService.php [hunk 1]
```diff
@@ -36,6 +38,7 @@ public function store($data)
         $applicant->graduation_year = $data['graduation_year'] ?? null;
         $applicant->university = $data['university_name'] ?? null;
         $applicant->centre_id = $data['centre'];
+        Mail::to($data['email_id'], $data['first_name'])->send(new CodetrekMailApplicant());
```
Comment (Ayush8923) [inline Modules/CodeTrek/Services/CodeTrekService.php:41]: Can you use the mail queue here?
Something like this.

```suggestion
        Mail::queue(new CodetrekMailApplicant($data));
```

File: Modules/CodeTrek/Emails/CodetrekMailApplicant.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Create a new message instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
```
Comment (Ayush8923) [inline Modules/CodeTrek/Emails/CodetrekMailApplicant.php:21]: Please update here accordingly.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comments. 
Please check.

File: Modules/CodeTrek/Emails/CodetrekMailApplicant.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+{
+    use Queueable, SerializesModels;
+    public $data;
+
+    /**
+     * Create a new message instance.
+     *
+     * @return void
+     */
+    public function __construct($data)
```
Comment (Ayush8923) [inline Modules/CodeTrek/Emails/CodetrekMailApplicant.php:19]: Please update all the places accordingly.
```suggestion
    public function __construct($Applicant)
```

File: Modules/CodeTrek/Resources/views/mail/codetrek.blade.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<div>
+	<style>
+		.line {
+			line-height: 1px;
+		}
+	</style>
+	<p>Hi, {{ $data['first_name'] }}</p>
+    <br>
+	<p>We're delighted to have you join Codtreak! Get ready to embark on an exciting coding journey filled with learning, challenges, and a supportive community. Feel free to explore our resources and reach out if you need any assistance.</p>
```
Comment (Ayush8923) [inline Modules/CodeTrek/Resources/views/mail/codetrek.blade.php:9]: We need to change this mail template. I will update you once I created the mail template for this.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some small comments.
Please take a look once.

**Comment by Ayush8923** [comment]
@Akshiteshnaithani Here is the Mail template.

```
Hello |*applicant_name*|,

We hope this email finds you well. On behalf of the entire team at ColoredCow, We wanted to take a moment to congratulate you on the commencement of your internship. We are thrilled to have you on board and excited to see the valuable contributions you will bring to our organization.

Once again, congratulations on starting your internship! We believe you have the potential to achieve great things, and we look forward to witnessing your growth and accomplishments over the course of your internship. 

Wishing you a successful and fulfilling internship journey!

Best regards,
ColoredCow
```

**Comment by Ayush8923** [comment]
> @Akshiteshnaithani Here is the Mail template.
> 
> ```
> Hello |*applicant_name*|,
> 
> We hope this email finds you well. On behalf of the entire team at ColoredCow, We wanted to take a moment to congratulate you on the commencement of your internship. We are thrilled to have you on board and excited to see the valuable contributions you will bring to our organization.
> 
> Once again, congratulations on starting your internship! We believe you have the potential to achieve great things, and we look forward to witnessing your growth and accomplishments over the course of your internship. 
> 
> Wishing you a successful and fulfilling internship journey!
> 
> Best regards,
> ColoredCow
> ```

@GauravGusain98  Also please take a look at this mail. And let me know if there are any changes needed in this mail template.

**Comment by Ayush8923** [comment]
> @Akshiteshnaithani Here is the Mail template.
> 
> ```
> Hello |*applicant_name*|,
> 
> We hope this email finds you well. On behalf of the entire team at ColoredCow, We wanted to take a moment to congratulate you on the commencement of your CodeTrek internship. We are thrilled to have you on board and excited to see the valuable contributions you will bring to our organization.
> 
> Once again, congratulations on starting your CodeTrek internship! You have the potential to achieve great things, and we look forward to witnessing your growth and accomplishments over the course of your internship. 
> 
> Wishing you a successful and fulfilling CodeTrek internship journey!
> 
> Best regards,
> ColoredCow
> ```

@GauravGusain98  You also please take a look at this mail. And let me know if there are any changes needed in this mail template.

**Comment by Akshiteshnaithani** [comment]
**After adding the applicant to Codetrek**

image

**The mail has been send to the applicant which is added to Codetrek**

image

File: Modules/CodeTrek/Services/CodeTrekService.php [hunk 5]
```diff
@@ -36,6 +38,7 @@ public function store($data)
         $applicant->graduation_year = $data['graduation_year'] ?? null;
         $applicant->university = $data['university_name'] ?? null;
         $applicant->centre_id = $data['centre'];
+        Mail::queue(new CodetrekMailApplicant($data));
```
Comment (Ayush8923) [inline Modules/CodeTrek/Services/CodeTrekService.php:41]: Could you please comment on this line for now?  For now, we don't need this mail functionality. This mail functionality is enabled when the use of the codeTrek module starts in the proper way.

Also could you please add this line before this line?
`This line will be uncommented in the future when the use of the codeTrek module starts in the proper way.`
```suggestion
        // Mail::queue(new CodetrekMailApplicant($data));
```

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
@Akshiteshnaithani Added one comment. Please implement this ASAP. Then I will merge this PR.

**Review by Ayush8923 — APPROVED** [review]
LGTM 👍

**Pull request merged**